### PR TITLE
Add annotation to network policy for invalid license error

### PR DESCRIPTION
**Pull request by TaoZou1**
Add NO_DFW_LICENSE to network policy annotation when return error code is 505
Also check return error from NSX side if it's an invalid license error

Test Done:
Case 1:
1. reboot the nsx-operator
2. delete the license from nsx
3. create the network policy to check if there is annotation
4. check if nsx-operator restart
Case 2:
1. based on case 1
2. update the license of NSX
3. wait for some time
4. check if annotation has been removed

### CODE REVIEW

**Review by timdengyun — APPROVED** [review]
LGTM

File: pkg/controllers/securitypolicy/securitypolicy_controller.go [hunk 1]
```diff
@@ -153,34 +182,21 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		if err := r.Service.CreateOrUpdateSecurityPolicy(realObj); err != nil {
 			if errors.As(err, &nsxutil.RestrictionError{}) {
 				log.Error(err, err.Error(), "securitypolicy", req.NamespacedName)
+				setSecurityPolicyErrorAnnotation(ctx, realObj, r.Client, "NO_DFW_LICENSE")
```
Comment (zhengxiexie) [inline pkg/controllers/securitypolicy/securitypolicy_controller.go:185]: define a const for NO_DFW_LICENSE ?

File: pkg/controllers/networkpolicy/networkpolicy_controller.go [hunk 2]
```diff
@@ -65,6 +65,35 @@ func deleteSuccess(r *NetworkPolicyReconciler, _ context.Context, o *networkingv
...
+func cleanNetworkPolicyErrorAnnotation(ctx context.Context, networkPolicy *networkingv1.NetworkPolicy, client client.Client) {
+	if networkPolicy.Annotations == nil {
+		return
+	}
+	if _, exists := networkPolicy.Annotations[common.NSXOperatorError]; exists {
+		delete(networkPolicy.Annotations, common.NSXOperatorError)
+	}
+	updateErr := client.Update(ctx, networkPolicy)
+	if updateErr != nil {
+		log.Error(updateErr, "Failed to clean NetworkPolicyannotation")
```
Comment (zhengxiexie) [inline pkg/controllers/networkpolicy/networkpolicy_controller.go:92]: NetworkPolicyAnnotation is better?

Comment (TaoZou1) [reply pkg/controllers/networkpolicy/networkpolicy_controller.go:92]: done

**Review by zhengxiexie — APPROVED** [review]
LGTM

**Review by timdengyun — APPROVED** [review]
LGTM

**Pull request merged**